src/modules/search/components/searchbar/index.tsx

```diff
@@ -1,33 +1,91 @@
 'use client';
-import React, { useState, FormEvent } from 'react';
 
-const SearchBar: React.FC = () => {
+import { useDictionary } from '@lib/context/dictionary-context';
+import React, { useState, FormEvent, useEffect, Fragment } from 'react';
+import { ProductPreviewType } from 'types/global';
+import { getProductsList } from '@lib/data/ecommerce';
+import SearchResult from '../search-result';
+import { Transition } from '@headlessui/react';
+import { useDebounce } from '@lib/hooks/use-debounce';
+
+const SearchBar = () => {
+  const dictionary = useDictionary();
+  const [searchDropdownOpen, setSearchDropdownOpen] = useState<boolean>(false);
   const [searchTerm, setSearchTerm] = useState<string>('');
+  const [searchResults, setSearchResults] = useState<ProductPreviewType[]>([]);
+  const debouncedSearch = useDebounce(searchTerm, 400);
+
+  const open = () => setSearchDropdownOpen(true);
+  const close = () => setSearchDropdownOpen(false);
+
+  const MAX_SEARCH_RESULTS = 6;
+
+  useEffect(() => {
+    const search = async () => {
+      const { response, nextPage, queryParams } = await getProductsList({
+        locale: 'fin',
+        queryParams: {
+          q: searchTerm,
+          limit: MAX_SEARCH_RESULTS,
+          is_giftcard: false,
+        },
+      });
+      setSearchResults(response.products);
+    };
+
+    search();
```

Comment:
Using async await inside useEffect is fine. But it can cause some confusion if there are any side effects happen before the function call. You can write plain then/catch block to avoid this confusion, or use the new <Suspense> component to handle the loading state and data fetching.
```suggestion
      getProductsList({
        locale: 'fin',
        queryParams: {
          q: searchTerm,
          limit: MAX_SEARCH_RESULTS,
          is_giftcard: false,
        },
      }).then(({response})=> setSearchResults(response.products));
```